src/resources/customers/mandates/CustomersMandatesResource.ts

```diff
@@ -0,0 +1,189 @@
+   *
+   * @returns
+   *
+   * @since 1.2.0
+   *
+   * @see https://docs.mollie.com/reference/v2/mandates-api/create-mandate
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public create(parameters: CreateParameters): Promise<Mandate>;
+  public create(parameters: CreateParameters, callback: Callback<Mandate>): void;
+  public create(parameters: CreateParameters) {
+    if (renege(this, this.create, ...arguments)) return;
+    const customerId = this.getParentId(parameters.customerId);
+    if (customerId == undefined || !checkId(customerId, 'customer')) {
+      throw new ApiError('The customer id is invalid');
+    }
+    const { customerId: _, ...data } = parameters;
+    return this.network.post(this.getResourceUrl(customerId), data);
+  }
+
+  /**
+   * Get a customer mandate by ID
+   *
+   * @param id - customers_mandate id
+   * @param params - Get customer mandate parameters
+   *                 (DEPRECATED SINCE 3.0.0) Can also be a callback function
+   * @param cb - (DEPRECATED SINCE 3.0.0) Callback function, can be used instead of the returned `Promise` object
+   *
+   * @returns Customer mandate
+   *
+   * @since 1.2.0
+   *
+   * @see https://docs.mollie.com/reference/v2/mandates-api/get-mandate
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public get(id: string, parameters: GetParameters): Promise<Mandate>;
+  public get(id: string, parameters: GetParameters, callback: Callback<Mandate>): void;
+  public get(id: string, parameters: GetParameters) {
+    if (renege(this, this.get, ...arguments)) return;
+    if (!checkId(id, 'mandate')) {
+      throw new ApiError('The customers_mandate id is invalid');
+    }
+    // parameters || {} is used here, because in case withParent is used, parameters could be omitted.
+    const customerId = this.getParentId((parameters || {}).customerId);
+    if (undefined == customerId || !checkId(customerId, 'customer')) {
+      throw new ApiError('The customer id is invalid');
+    }
+    const { customerId: _, ...query } = parameters || {};
+    return this.network.get(`${this.getResourceUrl(customerId)}/${id}`, query);
+  }
+
+  /**
+   * Get all of a customer's mandates
+   *
+   * @param params List mandates parameters
+   *               (DEPRECATED SINCE 3.0.0) Can also be a callback function
+   * @param cb - (DEPRECATED SINCE 3.0.0) Callback function, can be used instead of the returned `Promise` object
+   *
+   * @returns A list of found mandates
+   *
+   * @since 3.0.0
+   *
+   * @see https://docs.mollie.com/reference/v2/mandates-api/list-mandates
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public list(parameters: ListParameters): Promise<List<Mandate>>;
+  public list(parameters: ListParameters, callback: Callback<List<Mandate>>): void;
+  public list(parameters: ListParameters) {
+    if (renege(this, this.list, ...arguments)) return;
```

Comment:
In our internal codebase, we don't allow if-statements like this. We prefer always writing full if-statements.

Reply:
Personally, I don't like if-statements without the curly brackets either. However, this is very much boilerplace code. If we don't allow an exception here, this goes from one line to three lines **in every endpoint**.

Another option would be to wrap every endpoint. However, I don't think we could do overloading in that way.